eth/downloader/downloader.go

```diff
@@ -990,12 +996,28 @@ func (d *Downloader) findAncestor(p *peer) (uint64, error) {
 			// Make sure the peer actually gave something valid
 			headers := packet.(*headerPack).headers
 			if len(headers) == 0 {
-				glog.V(logger.Debug).Infof("%v: empty head header set", p)
+				glog.V(logger.Warn).Infof("%v: empty head header set", p)
 				return 0, errEmptyHeaderSet
 			}
+			// Make sure the peer's reply conforms to the request
+			for i := 0; i < len(headers); i++ {
```

Comment:
I think the following would be cleaner:

`for i := 0; i < int64(len(headers)); i++ {`

and remove the `int64(i)` casts.